legal-api/src/legal_api/resources/v2/request_tracker.py

```diff
@@ -30,11 +31,11 @@
 @jwt.has_one_of_roles([UserRoles.admin_edit, UserRoles.bn_edit])
 def get_bn_request_trackers(identifier: str):
     """Return a list of request trackers."""
-    legal_entity = LegalEntity.find_by_identifier(identifier)
-    if legal_entity is None:
+    business = business_service.fetch_business(identifier)
+    if business is None:
         return ({"message": "A valid business is required."}, HTTPStatus.BAD_REQUEST)
 
-    request_trackers = RequestTracker.find_by(legal_entity.id, RequestTracker.ServiceName.BN_HUB)
+    request_trackers = RequestTracker.find_by(business.id, RequestTracker.ServiceName.BN_HUB)
```

Comment:
This won't work as is when the business is a SP.  

When the business is a SP, ` business_service.fetch_business(identifier)` will return an instance of `alternate_name` which means that `RequestTracker.find_by` will return an incorrect request tracker instance or return nothing.  `RequestTracker.find_by` needs to be searching by `alternate_name_id` in this instance.

You should be doing something similar to this:

https://github.com/bcgov/lear/blob/feature-legal-name/legal-api/src/legal_api/models/filing.py#L865-L873

I suspect there are a bunch of other models you'll need to do this for.  Off the top of my head, I know that `documents`, `filings`, `comments`, `request_tracker` and `offices` are tied to `alternate_names` for SPs.